File: AdventOfCode.Console/Infra/InputHelper.cs
namespace AdventOfCode.Console.Infra;

public static class InputHelper
{
    public static IEnumerable<string> GetInputLines(byte day, byte part, bool testInput = false)
    {
        var path = $"./Inputs/d{day}p{part}{(testInput ? ".test" : "")}.txt";
        if (!File.Exists(path))
        {
            path = $"./Inputs/d{day}{(testInput ? ".test" : "")}.txt";
        }
        
        return File.ReadLines(path);
    }
}

File: AdventOfCode.Runner/InputHelper.fs
module AdventOfCode.Runner.InputHelper

open System.IO

let getInput day part testRun =
    let testSuffix = if testRun then ".test" else ""
    let inputLocation = $"./Inputs/d{day}p{part}{testSuffix}.txt"
    if File.Exists inputLocation then
        File.ReadLines inputLocation
    else
        File.ReadLines $"./Inputs/d{day}{testSuffix}.txt"